### ISSUE TITLE
How to use and import external components

### ISSUE THREAD

**Issue by fvanwijk** [ben-rogerson/twin.macro#153]
I'm developing a component library in a separate repo and for the time being I have it `npm link`ed in my application repository.

The component library is set up the default way using styled-components. I have one Button component which renders find in Storybook.
In the application repo I have this:

```ts
// MyComponent.tsx
import React from 'react';
import 'twin.macro';
import { Button } from 'my-component-lib';

export const MyComponent = () => <div tw="p-4"><Button>Text</Button/></div>;
```

The application is compiling fine, but in the browser console I get warnings and errors:
![image](https://user-images.githubusercontent.com/963136/93601553-a924ff80-f9c1-11ea-97c9-4efb08938d8f.png)

I'm not sure if this is a styled-component or a twin.macro issue.
What I'm sure about is that I don't know how to proceed.

Since the component lib repo needs the dependencies for standalone rendering (in storybook), I cannot just remove them.
Is it possible to 'reuse' the styled-component package when importing from my linked npm component-library package?

**Comment by ben-rogerson** [comment]
Hey Frank, I'm fairly sure this is styled-components thing.

Take a look at [this thread](https://github.com/styled-components/styled-components/pull/1102
) and the ones linked off it at the end.
Looks like you may need to use `StyleSheetManager` in your component library.

**Comment by fvanwijk** [comment]
You are right, it is not a Twin issue, so I will close it.
Thanks for the efforts nevertheless.

After some research I found out that when the module is linked, also the node_modules are included as it is just a symlink. That causes errors when import `styled_component` both from `node_modules` and `node_modules/my-component-lib/node_modules`.
I resolved it by setting an explicit webpack (babel-loader) alias to `node_modules/styled-components` but then I got more errors from other dependencies.

I finally concluded that this setup may cause more (webpack) issues in the future, so for now I moved all component-lib files into the application repo and when it is mature enough, we move it back into the other repo and publish via npm (without the node_modules)
